Gems/LmbrCentral/Code/Source/Shape/BoxShape.cpp

```diff
@@ -79,39 +79,53 @@ namespace LmbrCentral
 
     void BoxShape::InvalidateCache(InvalidateShapeCacheReason reason)
     {
-        m_intersectionDataCache.InvalidateCache(reason);
+        {
+            AZStd::unique_lock lock(m_mutex);
+            m_intersectionDataCache.InvalidateCache(reason);
+        }
     }
 
     void BoxShape::OnTransformChanged(const AZ::Transform& /*local*/, const AZ::Transform& world)
     {
-        m_currentTransform = world;
-        m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::TransformChange);
+        {
+            AZStd::unique_lock lock(m_mutex);
+            m_currentTransform = world;
+            m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::TransformChange);
+        }
         ShapeComponentNotificationsBus::Event(
             m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
             ShapeComponentNotifications::ShapeChangeReasons::TransformChanged);
     }
 
     void BoxShape::OnNonUniformScaleChanged(const AZ::Vector3& scale)
     {
-        m_currentNonUniformScale = scale;
-        m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+        {
+            AZStd::unique_lock lock(m_mutex);
+            m_currentNonUniformScale = scale;
+            m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+        }
         ShapeComponentNotificationsBus::Event(
             m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
             ShapeComponentNotifications::ShapeChangeReasons::ShapeChanged);
     }
 
     void BoxShape::SetBoxDimensions(const AZ::Vector3& dimensions)
     {
-        m_boxShapeConfig.m_dimensions = dimensions;
-        m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+        {
+            AZStd::unique_lock lock(m_mutex);
+            m_boxShapeConfig.m_dimensions = dimensions;
+            m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+        }
         ShapeComponentNotificationsBus::Event(
             m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
             ShapeComponentNotifications::ShapeChangeReasons::ShapeChanged);
     }
 
     AZ::Aabb BoxShape::GetEncompassingAabb()
     {
-        m_intersectionDataCache.UpdateIntersectionParams(m_currentTransform, m_boxShapeConfig, m_currentNonUniformScale);
+        AZStd::shared_lock lock(m_mutex);
+        m_intersectionDataCache.UpdateIntersectionParams(m_currentTransform, m_boxShapeConfig, m_mutex, m_currentNonUniformScale);
```

Comment:
Is the `m_intersectionDataCache` being modified by the `UpdateIntersectionParams` function or is it a const member?

Reply:
UpdateIntersectionParams will check to see if the cache needs to be updated, and then update it if it does. To avoid needing unique locks in the common case where the cache is up-to-date, this passes in the mutex, and UpdateIntersectionParams() will change it from a shared_lock to a unique_lock while updating the cache, and then back to a shared_lock before exiting.